Sources/SourceKitLSP/Swift/MacroExpansionReferenceDocumentURLData.swift

```diff
@@ -15,29 +15,34 @@ import LanguageServerProtocol
 import RegexBuilder
 
 /// Represents url of macro expansion reference document as follows:
-/// `sourcekit-lsp://swift-macro-expansion/LaCb-LcCd.swift?primaryFilePath=&fromLine=&fromColumn=&toLine=&toColumn=&bufferName=`
+/// `sourcekit-lsp://swift-macro-expansion/LaCb-LcCd.swift?primaryFilePath=&fromLine=&fromColumn=&toLine=&toColumn=&bufferName=&parentReferenceDocumentURL=`
 ///
 /// Here,
 ///  - `LaCb-LcCd.swift`, the `displayName`, represents where the macro will expand to or
 ///    replace in the source file (i.e. `macroExpansionEditRange`)
 ///  - `primaryFilePath` denoting the URL of the source file
 ///  - `fromLine`, `fromColumn`, `toLine`, `toColumn` represents the cursor's `selectionRange`
 ///  - `bufferName` denotes the buffer name of the specific macro expansion edit
+///  - `parentReferenceDocumentURL` the reference document url of the macro
+///    which expands to the current macro (optional)
 package struct MacroExpansionReferenceDocumentURLData {
   package static let documentType = "swift-macro-expansion"
 
   package var primaryFileURL: URL
+  package var parentReferenceDocumentURL: URL?
```

Comment:
Yeah, I was focusing on limiting the usage of the `parentReferenceDocumentURL` and brought its count to just two places where i reference it as marked by `// usage`, I thought it was finally a good fit to use the bufferName directly instead of the url, but I decided to optimise everything after I finish the initial work.